pkg/kubelet/util/util_windows.go

```diff
@@ -123,3 +131,37 @@ func GetBootTime() (time.Time, error) {
 	}
 	return currentTime.Add(-time.Duration(output) * time.Millisecond).Truncate(time.Second), nil
 }
+
+type reparseDataBufferHeader struct {
+	ReparseTag        uint32
+	ReparseDataLength uint16
+	Reserved          uint16
+}
+
+type reparseDataBuffer struct {
+	Header reparseDataBufferHeader
+	Detail [syscall.MAXIMUM_REPARSE_DATA_BUFFER_SIZE]byte
+}
+
+// IsUnixDomainSocket returns whether a given file is a AF_UNIX socket file
+func IsUnixDomainSocket(fileInfo os.FileInfo, filePath string) (bool, error) {
+	fd, err := windows.CreateFile(windows.StringToUTF16Ptr(filePath), windows.GENERIC_READ, 0, nil, windows.OPEN_EXISTING, windows.FILE_FLAG_OPEN_REPARSE_POINT|windows.FILE_FLAG_BACKUP_SEMANTICS, 0)
+	if err != nil {
+		return false, errors.Wrap(err, "CreateFile failed")
+	}
+	defer windows.CloseHandle(fd)
+
+	rdbbuf := make([]byte, syscall.MAXIMUM_REPARSE_DATA_BUFFER_SIZE)
+	var bytesReturned uint32
+	if err = windows.DeviceIoControl(fd, windows.FSCTL_GET_REPARSE_POINT, nil, 0, &rdbbuf[0], uint32(len(rdbbuf)), &bytesReturned, nil); err != nil {
+		if err.Error() == msgNotAReparsePoint {
+			return false, nil
+		}
+		return false, errors.Wrap(err, "FSCTL_GET_REPARSE_POINT failed")
+	}
+	rdb := (*reparseDataBuffer)(unsafe.Pointer(&rdbbuf[0]))
+	if rdb.Header.ReparseTag == reparseTagSocket {
+		return true, nil
+	}
+	return false, nil
+}
```

Comment:
Does this return false or an error when the argument is a named pipe?

I'm not sure if that question even totally makes sense, I just know that named pipes are the other mechanism for IPC with Windows Kubernetes and could perhaps see someone mistakenly passing a named pipe path to this method.

Reply:
For a named pipe, IsUnixDomainSocket returns `false` along with a non-nil err as it gets treated similar to an invalid file. Added a unit-test with a named pipe to clarify/check the behavior.